runtime/src/main/java/org/corfudb/workflow/Step.java

```diff
@@ -0,0 +1,78 @@
+package org.corfudb.workflow;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import lombok.Getter;
+
+/**
+ * Independent process to be executed with a set of parameters.
+ * A series of sequential steps form a workflow.
+ *
+ * <p>Created by zlokhandwala on 10/16/17.
+ */
+public class Step<T, R> {
+
+    /**
+     * Unique step Id.
+     */
+    public final UUID stepId = UUID.randomUUID();
```

Comment:
Why should this be a UUID rather than just a integer? If an integer, it could be ordered.